cukes-rest/src/main/java/lv/ctco/cukesrest/di/SingletonObjectFactory.java

```diff
@@ -0,0 +1,77 @@
+package lv.ctco.cukesrest.di;
+
+import com.google.common.collect.Sets;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.Stage;
+import cucumber.api.guice.CucumberModules;
+import cucumber.api.java.ObjectFactory;
+import cucumber.runtime.java.guice.ScenarioScope;
+import lv.ctco.cukesrest.internal.di.CukesGuiceModule;
+
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkState;
+
+public class SingletonObjectFactory implements ObjectFactory {
+
+    private static final Set<Module> MODULES = Sets.newConcurrentHashSet();
+
+    static {
+        MODULES.add(CucumberModules.SCENARIO);
+        MODULES.add(new CukesGuiceModule());
+    }
+
+    private static Injector injector = null;
+
+    /**
+     * Internal use only.
+     * <br><br>
+     * Use {@link SingletonObjectFactory#instance()} instead.
+     */
```

Comment:
Was thinking the same except I wasn't fully convinced since it's not true and that constructor will never go away but I'm okay adding it since there is value.